spec/requests/api/v1/carts/remove_product_spec.rb

```diff
@@ -60,5 +56,13 @@
 
       it_behaves_like 'response bad_request'
     end
+
+    context 'when user not logged in' do
+      let(:headers) { nil }
+      let(:quantity) { 1 }
+      let(:product) { create(:product) }
+
+      it_behaves_like 'user is not logged in'
```

Comment:
Personally I don't use shared examples too much. Sometimes they make tests less descriptive and are not so useful for documentation.

I'd suggest to rename you shared_examples 
```suggestion
      it_behaves_like 'an unauthorised request'
```

https://www.betterspecs.org/#shared

Reply:
updated.